Tests/Common/Securities/BuyingPowerModelTests.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Tests.Common.Securities
+{
+    class BuyingPowerModelTests
+    {
+        // The only instance in which this function is used is when
+        // (Math.Abs(currentOrderMargin) > Math.Abs(targetMargin))
+        [TestCase(-1000, -900, 25)]
+        [TestCase(-900, -880, 25)]
+        [TestCase(1200, 1050, 25)]
+        [TestCase(1225, 1212, 25)]
+        public void OrderAdjustmentCalculation(decimal currentOrderMargin, decimal targetMargin, decimal perUnitMargin)
```

Comment:
Cool! Let's add a few more cases/asserts
- Where `currentOrderMargin` is less than `targetMargin`
- Where current and/or target margin/ perUnitMargin aren't already rounded values of the lot size, could even play with the lot size value
- Let's assert the resulting order quantity too, so it's explicit